src/main/java/games/strategy/triplea/attachments/AbstractConditionsAttachment.java

```diff
@@ -377,5 +377,5 @@ public void changeChanceDecrementOrIncrementOnSuccessOrFailure(final IDelegateBr
   }
 
   @Override
-  public void validate(final GameData data) throws GameParseException {}
+  public abstract void validate(final GameData data) throws GameParseException;
```

Comment:
Since all descendants implement this method, I think you can just remove this override, as it will be inherited from the superclass.

Reply:
Correct, didn't notice that.